test/recipes/70-test_tls13messages.t

```diff
@@ -296,4 +301,57 @@ checkhandshake($proxy, checkhandshake::HRR_RESUME_HANDSHAKE,
                | checkhandshake::PSK_CLI_EXTENSION
                | checkhandshake::PSK_SRV_EXTENSION,
                "Resumption handshake with HRR test");
+
+#Test 16: signature_algorithms with 1.3-only ClientHello
+my $testtype = PURE_SIGALGS;
+$proxy->clear();
+$proxy->clientflags("-tls1_3");
+$proxy->filter(\&modify_sigalgs_filter);
+$proxy->start();
+
+#Test 17: signature_algorithms with backwards compatible ClientHello
+$testtype = COMPAT_SIGALGS;
+$proxy->clear();
+$proxy->filter(\&modify_sigalgs_filter);
+$proxy->start();
 unlink $session;
+
+sub modify_sigalgs_filter
+{
+    my $proxy = shift;
+
+    # We're only interested in the initial ClientHello
+    return if ($proxy->flight != 0);
+
+    foreach my $message (@{$proxy->message_list}) {
+        my $ext;
+        my @algs;
+
+        if ($message->mt == TLSProxy::Message::MT_CLIENT_HELLO) {
+            if ($testtype == PURE_SIGALGS) {
+                $ext = $message->get_extension(TLSProxy::Message::EXT_SIG_ALGS);
+                @algs = unpack('S>*', $ext);
+                # unpack will unpack the length as well
+                shift @algs;
+                foreach (@algs) {
+                    if ($_ == 0x0402 || $_ == 0x0502 || $_ == 0x0602) {
+                        die "DSA/SHA2 sigalg sent for 1.3-only ClientHello\n"
```

Comment:
Don't die in a test. That will stop all the remaining tests from running. Set some global variable with the result and then check it afterwards with an "ok()" call.